packages/renderer/src/lib/preferences/PreferencesConnectionDetailsTerminal.svelte

```diff
@@ -0,0 +1,187 @@
+<script lang="ts">
+import '@xterm/xterm/css/xterm.css';
+
+import type { ProviderConnectionShellDimensions, ProviderConnectionStatus } from '@podman-desktop/api';
+import { EmptyScreen } from '@podman-desktop/ui-svelte';
+import { FitAddon } from '@xterm/addon-fit';
+import { SerializeAddon } from '@xterm/addon-serialize';
+import { Terminal } from '@xterm/xterm';
+import { onDestroy, onMount } from 'svelte';
+import { router } from 'tinro';
+
+import { getExistingTerminal, registerTerminal } from '/@/stores/provider-terminal-store';
+import type { ProviderContainerConnectionInfo, ProviderInfo } from '/@api/provider-info';
+
+import { TerminalSettings } from '../../../../main/src/plugin/terminal-settings';
+import { getTerminalTheme } from '../../../../main/src/plugin/terminal-theme';
+import NoLogIcon from '../ui/NoLogIcon.svelte';
+
+interface ProviderDetailsTerminalProps {
+  provider: ProviderInfo;
+  connectionInfo: ProviderContainerConnectionInfo;
+  screenReaderMode?: boolean;
+}
+
+let { provider, connectionInfo, screenReaderMode = false }: ProviderDetailsTerminalProps = $props();
+let terminalXtermDiv: HTMLDivElement;
+let shellTerminal: Terminal;
+let currentRouterPath: string;
+let sendCallbackId: number | undefined;
+let terminalContent: string = '';
+let serializeAddon: SerializeAddon;
```

Comment:
question: don't they need to be $state ?

Reply:
Checks are passing so I would say that they don't have to be